### PR TITLE
Add fee market slash event

### LINKED ISSUE #1009: Add slash event in the fee-market pallet
Add some events to explain why the slash occurred.

### PR DESCRIPTION
**Pull request by AsceticBear**
Close #1009 

After discussing with Yalin, both s2s message and fee market pallets will clean the old order/message data as time goes by. So it is better to keep all these fields about the order status.

### CODE REVIEW

File: primitives/fee-market/src/lib.rs [hunk 1]
```diff
@@ -213,6 +213,42 @@ where
...
+/// The detail information about slash behavior
+#[derive(Clone, Encode, Decode, Eq, PartialEq, Debug)]
+pub struct SlashReport<AccountId, BlockNumber, Balance> {
+	pub lane: LaneId,
+	pub message: MessageNonce,
+	pub sent_time: BlockNumber,
+	pub confirm_time: Option<BlockNumber>,
+	pub delay_time: Option<BlockNumber>,
+	pub account_id: AccountId,
+	pub amount: Balance,
```
Comment (AsceticBear) [inline primitives/fee-market/src/lib.rs:225]: After discussing with Yalin, both s2s message and fee market pallets will clean the old order/message data as time goes by. So it is better to keep all these fields.

File: primitives/fee-market/src/lib.rs [hunk 2]
```diff
@@ -213,6 +213,42 @@ where
...
+	pub fn new(
+		order: &Order<AccountId, BlockNumber, Balance>,
+		account_id: AccountId,
+		amount: Balance,
+	) -> Self {
+		Self {
+			lane: order.lane,
+			message: order.message,
+			sent_time: order.sent_time,
+			confirm_time: order.confirm_time,
```
Comment (HackFisher) [inline primitives/fee-market/src/lib.rs:244]: LGTM since we do not have an NewOrder event on hand for indexing.

**Review by HackFisher — APPROVED** [review]

File: frame/fee-market/src/lib.rs [hunk 3]
```diff
@@ -111,6 +111,8 @@ pub mod pallet {
 		UpdateCollateralSlashProtect(RingBalance<T>),
 		/// Update market assigned relayers numbers. \[new_assigned_relayers_number\]
 		UpdateAssignedRelayersNumber(u32),
+		/// Slash report
+		FeeMarketSlash(SlashReport<T::AccountId, T::BlockNumber, RingBalance<T>>),
```
Comment (AurevoirXavier) [inline frame/fee-market/src/lib.rs:115]: Metadata: T::BlockNumber

Comment (AsceticBear) [reply frame/fee-market/src/lib.rs:116]: Done

**Comment by AsceticBear** [comment]
@HackFisher @AurevoirXavier Can we merge this today?

**Comment by AurevoirXavier** [comment]
Do you want to release a new runtime version after this?

**Comment by AsceticBear** [comment]
This is not an urgent merge request. 

I wish to create a new release after https://github.com/darwinia-network/runtime-overrides/pull/5 as soon as possible.  I promise the docs team that I will complete tracing the writing task this week.

Do we have a plan to upgrade pangolin this week?

**Comment by AsceticBear** [comment]
Waiting for substrate upgrade merged, then no need to update type.json.

**Review by AurevoirXavier — APPROVED** [review]

**Pull request merged**